engine/lib/elgglib.php

```diff
@@ -1426,7 +1426,31 @@ function _elgg_normalize_plural_options_array($options, $singulars) {
  */
 function _elgg_shutdown_hook() {
 	global $START_MICROTIME;
+	
+	elgg_set_config('shutdown_flag', true);
```

Comment:
I'd prefer adding a new function elgg_has_shutdown(). Better would be to move this logic to just before shutdown is triggered. That way, if you're handling shutdown, you know no more output is allowed.